src/dtemplate.d

```diff
@@ -4998,7 +4998,8 @@ private bool reliesOnTident(Type t, TemplateParameters* tparams = null, size_t i
         {
             //printf("CallExp.reliesOnTident('%s')\n", e.toChars());
             visit(cast(UnaExp)e);
-            if (!result)
+            // e.to can be null for `cast()`
+            if (!result && e.to)
```

Comment:
then maybe we can warn ?
`uint a = cast()12;` 
does indeed compile but I feel it's strange to allow that